### PR TITLE
Tweaks turret deployment

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Since https://github.com/ParadiseSS13/Paradise/pull/17662 was merged, turrets on certain admin shuttles have not been deploying since they are behind directional windows. These are not the same as fulltile windows and should not logically block turret deployment, as such I have implemented an exceptions whitelist to dense objects that would normally block their deployment. This currently includes all indoors and all directional windows.
Centcomm subclass turrets can also deploy regardless of density.

## Why It's Good For The Game
Admin turrets (Centcomm ones) should disregard normal rules since they are often placed in strange places that make it hard for players to kill them.
Also, logically, things that are not actually blocking the turret (directional windows) should be fine for allowing deployment.

## Changelog
:cl:
tweak: Tweaked admin turrets to always deploy when needed, regardless of location
fix: Fixed directional windows blocking regular turret deployment.
/:cl:

### CODE REVIEW

File: code/game/machinery/portable_turret.dm [hunk 1]
```diff
@@ -614,21 +616,35 @@ GLOBAL_LIST_EMPTY(turret_icons)
...
+		pop_up()
 	// check if anything's preventing us from raising
 	var/turf/T = get_turf(src)
 	for(var/atom/A in T)
 		if(A == src)
 			continue
 		if(A.density)
+			if(is_type_in_typecache(A, deployment_whitelist))
+				continue
 			return
```
Comment (moxian) [inline code/game/machinery/portable_turret.dm:644]: If you don't want to hardcode the types, you can do
```suggestion
		if(!A.CanPass(src, src.loc))
				return
```
This appears to work in my testing, with the exception that dense mobs (i.e. standing humans/bots) no longer prevent deployment.

Comment (S34NW) [reply code/game/machinery/portable_turret.dm:644]: Unsure if that's a bit of a scuffed solution, I'll get a PRR to have a look before implementing

**Review by SteelSlayer — APPROVED** [review]

File: code/game/machinery/portable_turret.dm [hunk 2]
```diff
@@ -66,6 +66,8 @@
 	var/scan_range = 7
 	var/always_up = FALSE		//Will stay active
 	var/has_cover = TRUE		//Hides the cover
+	/// Deployment override to allow turret popup on/under dense turfs/objects, for admin/CC turrets
+	var/deployment_override
```
Comment (AffectedArc07) [inline code/game/machinery/portable_turret.dm:70]: Initialize this as false if its meant to be a boolean state

Comment (S34NW) [inline code/game/machinery/portable_turret.dm:70]:
```suggestion
	var/deployment_override = FALSE
```

**Review by AffectedArc07 — APPROVED** [review]

File: code/game/machinery/portable_turret.dm [hunk 3]
```diff
@@ -614,21 +616,36 @@ GLOBAL_LIST_EMPTY(turret_icons)
...
+/obj/machinery/porta_turret/proc/check_popUp()
+	/// Whitelist to determine what objects can be put over turrets while letting them deploy
+	var/static/list/deployment_whitelist = typecacheof(list(
+	/obj/structure/window/reinforced,
+	/obj/structure/window/basic,
+	/obj/structure/window/plasmabasic,
+	/obj/structure/window/plasmareinforced,
+	/obj/machinery/door/window,
+	/obj/structure/railing,
+	))
```
Comment (AffectedArc07) [inline code/game/machinery/portable_turret.dm:628]:
```suggestion
	var/static/list/deployment_whitelist = typecacheof(list(
		/obj/structure/window/reinforced,
		/obj/structure/window/basic,
		/obj/structure/window/plasmabasic,
		/obj/structure/window/plasmareinforced,
		/obj/machinery/door/window,
		/obj/structure/railing,
	))
```

And this

File: code/game/machinery/portable_turret.dm [hunk 4]
```diff
@@ -614,21 +616,36 @@ GLOBAL_LIST_EMPTY(turret_icons)
 		if(target(M))
 			return 1
 
-/obj/machinery/porta_turret/proc/popUp()	//pops the turret up
+/obj/machinery/porta_turret/proc/check_popUp()
```
Comment (AffectedArc07) [inline code/game/machinery/portable_turret.dm:619]: Also if you renamed `popUp()` to `pop_up()` why not make this `check_pop_up()`

Comment (S34NW) [reply code/game/machinery/portable_turret.dm]: good pont well made

**Pull request merged**